pysr/sr.py

```diff
@@ -2586,3 +2583,190 @@ def _mutate_parameter(param_name: str, param_value):
         return False
 
     return param_value
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        self.recursive_history_length = recursive_history_length
+        super().__init__(**kwargs)
+
+    def fit(
+        self,
+        X,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+    ) -> "PySRRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_samples, 1) or (1, n_samples).
```

Comment:
```suggestion
            Training time series data of shape (n_times, 1).
```

Reply:
If this should have multiple features, I think it would be
```suggestion
            Training time series data of shape (n_times, n_features).
```
and the 1D scalar time series would just have `n_features=1`, but the user could pass multiple features if they wish.